### PR TITLE
Align BN trainable behaviour with TF 2.0

### PR DESCRIPTION
**Pull request by datumbox**
### Summary
TensorFlow 2.0 has changed the behavior of Batch Normalization to be in inference mode when the `trainable` parameter is set to False. Reference from the [documentation](https://www.tensorflow.org/api_docs/python/tf/keras/layers/BatchNormalization#output_shape_2):

> ...in the case of the BatchNormalization layer, setting trainable = False on the layer means that the layer will be subsequently run in inference mode... This behavior has been introduced in TensorFlow 2.0, in order to enable layer.trainable = False to produce the most commonly expected behavior in the convnet fine-tuning use case.

2 years ago, I have submitted a PR (#9965) that changed Keras' BN behaviour to what TF uses tofday but it was not merged. I submit it for reconsideration, hoping we can align the two behaviors.

I'm happy to make changes to the PR to see it getting merged to master.

### Related Issues

#9214 
#10014
#10045
#11803

### CODE REVIEW

**Comment by CMCDragonkai** [comment]
This only impacts people who are not using `tf.keras` right?
